OpenSim/Simulation/StatesTrajectory.h

```diff
@@ -321,6 +321,22 @@ class OSIMSIMULATION_API StatesTrajectory {
     };
 #endif
 
+    /** Thrown by createFromStatesStorage(). */
+    class ModelHasNoSystem : public Exception {
```

Comment:
This is a useful and more general Exception that could live with Model. Component has `ComponentHasNoSystem` but this is a good case for a Model specific one.